lib/sax/sax_ltx.js

```diff
@@ -69,13 +70,25 @@ var SaxLtx = module.exports = function SaxLtx() {
                     attrs = {}
                 }
                 break
+            case STATE_CDATA:
+                if(c === 93 /* ] */ && data.substr(pos + 1, 2) === ']>') {
+                    var cData = endRecording();
+                    this.emit('text', unescapeXml(cData));
+                    state = STATE_IGNORE_TAG;
+                }
+                break;
             case STATE_TAG_NAME:
                 if (c === 47 /* / */ && recordStart === pos) {
                     recordStart = pos + 1
                     endTag = true
                 } else if (c === 33 /* ! */ || c === 63 /* ? */) {
-                    recordStart = undefined
-                    state = STATE_IGNORE_TAG
+                    if(data.substr(pos + 1, 7) === '[CDATA[') {
+                        state = STATE_CDATA;
+                        recordStart = pos + 8;
+                    } else {
+                        recordStart = undefined
+                        state = STATE_IGNORE_TAG
+                    }
```

Comment:
@swissmanu 

> what do you think about the implementation? i dont like the way how i had to implement the look ahead for detecting CDATA at all. any other ideas?

I get the feeling but I like how it doesn't interfere too much with the rest of the code. :+1: and maybe we can come back to this in the future.